### DISCUSSION TITLE
Display climbing cliffs

### DISCUSSION THREAD

**Discussion by Scafir** [organicmaps/organicmaps#1259] [Ideas]
First of all, huge thanks to the devs for all your work, I really enjoy the app!

I would like to suggest adding outdoors climbing cliffs on the visuals (`natural=cliff`, `sport=climbing`).
My use case: many climbing cliffs require a somewhat long walking approach and are usually not indicated onsite (thus the need to use books or other means for directions). Not having this info on a map can be a bother when one has clear directions and a source of frustration when one doesn't. Moreover, it would enable easier planning (e.g: which area has cliffs easily accessible and thus, which book to buy). I also believe that it would not contribute to crowding the map much, since such cliffs are usually in remote locations.

Exemple of such cliffs: [cliff as a line, many tags](https://www.openstreetmap.org/edit#map=18/45.13583/13.68906), [point cliff, basic tags, area somewhat crowded](https://www.openstreetmap.org/edit#map=19/46.29509/6.97765)

**Comment by biodranik** [comment]
Thanks, it is a good idea. Do you have screenshots of these cliffs from other mobile apps? How do apps render them in a recognizable way?

**Reply by Scafir** [reply]
Apart from iD (osm browser editor), I do not know of any maps that provide this utility. At least not google, apple or bing.  Even dedicated climbing websites do not have anything fine grain. To my knowledge, it would be the first time a general purpose map application implements something like it!

**Reply by Scafir** [reply]
I have looked into it a little bit more. First of all, you can find some reference to already existing implementations in the osm [wiki](https://wiki.openstreetmap.org/wiki/Climbing), under "Renderer and applications". I found that [kletterspot](https://kletterspots.de/) had the most visually appealing rendering. Pictograms are also available [there](https://wiki.openstreetmap.org/wiki/Map_Icons/WikiProject_Pictograms.).

**Comment by Scafir** [comment]
I realized that it may be a bit more confusing to implement than I thought. You can find below my suggestions:
### Preface
Climbing geographic information can be a general location with the tag `climbing=area` (e.g a mountain), a specific zone `climbing=crag` or `climbing=boulder` (e.g a specific cliff) or a specific route with `climbing=route`. The icon could be the same

### Areas
All areas with the tag `sport=climbing` should be displayed as a centered point, with more priority than what's below. Moreover, an area with `climbing=area` has more importance than `climbing=crag` or `climbing=boulder`, which are equally important. Important info for areas is the `name`.

### Ways
If the way has `sport=climbing` and `natural=cliff`, then it should be displayed as a centered point. Important info is the `name`.
If the way has `climbing=route`, then it should be displayed as a point located at the node `climbing=route_bottom` that composes it. Important info is `name`, `climbing:length`. Difficulty could be added, but a bit too complicated right now (there are different standards).
All other ways can be ignored.

### Nodes
Display all nodes with a tag `sport=climbing`, which are not part of a way. If the tag is `climbing=route` or `climbing=route_bottom`, important information is  `name`, `climbing:length` (same story as ways for the difficulty). Otherwise, important information is just `name`.
